tutorials/platform/android/android_plugin.rst

```diff
@@ -1,165 +1,388 @@
+      [plugin]
 
-The Godot editor will automatically parse all ``.gdap`` files in the ``res://android/plugins`` directory and show a list of detected and toggleable plugins in the Android export presets window under the **Plugins** section.
+      name="<plugin_name>"
+      description="<plugin_description>"
+      author="<plugin_author>"
+      version="<plugin_version>"
+      script="<relative_path_to_the_editorplugin_tool_script>"
 
-In order to allow GDScript to communicate with your Java Singleton, you must annotate your function with ``@UsedByGodot``. The name called from GDScript must match the function name exactly. There is **no** coercing ``snake_case`` to ``camelCase``.
+For reference, here is the `folder structure for the Godot Android plugin project template <https://github.com/m4gr3d/Godot-Android-Plugin-Template/tree/main/plugin/export_scripts_template>`_.
+At build time, the contents of the ``export_scripts_template`` directory is copied to the ``addons/<plugin_name>`` directory:
 
-.. image:: img/android_export_preset_plugins_section.png
+.. code-block:: none
 
-From your script::
+    export_scripts_template/
+    |
+    +--.bin/                        # output location for the AAR binaries
+    |
+    +--.export/                     # tool scripts directory
+    |   |
+    |   +--editor_export_plugin.gd  # EditorExportPlugin tool script
+    |   |
+    |   +--editor_plugin.gd         # EditorPlugin tool script
+    |
+    +--plugin.cfg                   # plugin INI file
+
+
+Packaging a v2 Android plugin with GDExtension capabilities
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For GDExtension, we follow the same steps as for `Packaging a v2 Android plugin`_ and add the `GDExtension config file <https://docs.godotengine.org/en/stable/tutorials/scripting/gdextension/gdextension_cpp_example.html#using-the-gdextension-module>`_ in
+the same location as ``plugin.cfg``.
+
+For reference, here is the `folder structure for the GDExtension Android plugin project template <https://github.com/m4gr3d/GDExtension-Android-Plugin-Template/tree/main/plugin/export_scripts_template>`_.
+At build time, the contents of the ``export_scripts_template`` directory is copied to the ``addons/<plugin_name>`` directory:
+
+.. code-block:: none
+
+    export_scripts_template/
+    |
+    +--.bin/                        # output location for the AAR binaries
+    |
+    +--.export/                     # tool scripts directory
+    |   |
+    |   +--editor_export_plugin.gd  # EditorExportPlugin tool script
+    |   |
+    |   +--editor_plugin.gd         # EditorPlugin tool script
+    |
+    +--plugin.cfg                   # plugin INI file
+    |
+    +--plugin.gdextension           # GDExtension config file
+
+
+Here is what the ``plugin.gdextension`` config file should look like::
+
+    [configuration]
+
+    entry_symbol = "plugin_library_init"
+    compatibility_minimum = 4.2
```

Comment:
```suggestion
    compatibility_minimum = "4.2"
```
See https://github.com/godotengine/godot-docs/issues/7864

Reply:
Thanks for the info! I need to update the project templates as well.